[docs] Instruction to deal with v2 invalid license message (#5074)
mui/mui-x: docs/data/advanced-components/overview.md

## License key installation

When you purchase a commercial license, you'll receive your license key by email.
This key removes all watermarks and console warnings.

:::warning

> ⚠️ If you purchased your license **after May 13, 2022**, You'll obtain a **v2** license key.
> 
> To use the v2 licenses, you'll need to update the MUI X packages to the latest or at least the `5.11` version.
> 
> If updating is not practicable or viable, please contact our support at sales@mui.com

Reviewer: Here's how I would phrase this:

```suggestion
All licenses purchased after **May 13, 2022** are _v2_ licenses.

The v2 license key is only compatible with MUI X v5.11 or later.
Please update your packages if you are using an earlier version.

If this is not a viable solution for your use case, please contact us at sales@mui.com.
```

## License key installation

> When you purchase a commercial license, you'll receive your license key by email.

Author: Alright. That makes sense

This key removes all watermarks and console warnings.

:::warning

> Licenses purchased after **May 13, 2022** are only compatible with MUI X version `5.11` or later.

Author: ```suggestion
Licenses purchased after **May 13, 2022** are only compatible with MUI X `v5.11.0` or later.
```

This key removes all watermarks and console warnings.

:::warning

> ⚠️ Licenses purchased after **May 13, 2022** are only compatible with MUI X version `5.11` or later.

Reviewer 2: We decided that the `:::warning` sections did not need the icon (I can find the discussion on the core if you want)

Author: ```suggestion
Licenses purchased after **May 13, 2022** are only compatible with MUI X version `5.11` or later.
```

#### Invalid license key

This error indicates that your license key doesn't match what was issued by MUI (this is likely a typo).

#### Invalid license key (Error: extracting license expiry timestamp)

> This is a more specific situation of the error above. If additionally to the invalid license message, you see an error on the console that looks like:

Reviewer 3: "If additionally" is strange, maybe?

```suggestion
The following JavaScript exception indicates that you may be trying to validate [the new license's key format on an older version of the npm package](#license-key-installation).
```

:::error
Error extracting license expiry timestamp.

TypeError: Cannot read properties of null (reading '1') at verifyLicense.
:::

It indicates that you may be trying to validate [the new license's key format on an older package](#license-key-installation).

> In this case, please update to MUI X `5.11` or later.

Reviewer 2: ```diff
-to `v5.11.0` or later.
+to `v5.11.0` or above.
```

Not sure
